src/target/source/source_module.cc

```diff
@@ -191,17 +192,36 @@ class CSourceCrtMetadataModuleNode : public runtime::ModuleNode {
           << "}\n";
   }
 
+  void GenerateAOTDescriptor() {
+    code_ << "#include \"tvm/runtime/crt/internal/aot_executor/aot_executor.h\"\n";
```

Comment:
maybe should promote the tvm_model_t to a non-internal header though? I think we consider the operator code as external to CRT

Reply:
I am not entirely sure I follow :) . `tvm_model_t` is part of the internal interface to use AOT, and we agreed to move the executor interface to be internal for now. Also, why this would make the operator code internal?

Reply:
ah, right--we wanted to defer making a public API to a follow-on PR. that's still true, so perhaps we should keep this here for this PR. my thinking was that depending on a header file in `tvm/runtime/crt/internal` makes this code internal. however, when compiling µTVM project, the model operator is compiled separately from CRT, so the current compilation flow treats operator code as external right now. 

with that said, let's ignore this for now--as you said, we're keeping the AOT executor interface internal now, and we don't enforce this internal-external split e.g. by creating a separate include dir at build time now anyhow.